x-pack/plugins/ml/public/application/services/ml_server_info.ts

```diff
@@ -53,10 +53,11 @@ export function isCloud(): boolean {
 }
 
 export function getCloudDeploymentId(): string | null {
-  if (cloudInfo.cloudId === null) {
-    return null;
-  }
-  const tempCloudId = cloudInfo.cloudId.replace(/^.+:/, '');
+  return cloudInfo.cloudId === null ? null : extractDeploymentId(cloudInfo.cloudId);
+}
+
+export function extractDeploymentId(cloudId: string) {
+  const tempCloudId = cloudId.replace(/^(.+)?:/, '');
```

Comment:
nit: you can achieve it without regexp, `const tempCloudId = cloudId.split(':')[0]`

Reply:
if the deployment name isn't present at all then the array will only have a length of 1 and so an extra check will be needed.
i think using replace here is the simplest approach, even though it's not the most performant.

Reply:
But you regexp rely on `:` anyway. It expects something like`":xyz"`, so split will return 2 elements anyway, the first one will be an empty string. Not critical at all anyway, just suppose it's always easier without regexp